### PR TITLE
Chore:  Resolve "environment variable" notice in the github build artifacts

### PR DESCRIPTION
**Pull request by martinfrances107**
I saw this warning in the build artefacts 

>The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
test (windows-latest)

udeps is the only place in the codebase where we still use the deprecated mechanism.

### What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [x] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

File: .github/workflows/udeps.yml [hunk 1]
```diff
@@ -115,7 +115,7 @@ jobs:
           if [ "${UTILS}" == "true" ]; then crates[${#crates[@]}]="\"./core/tauri-utils\""; fi
           if [ "${BUNDLER}" == "true" ]; then crates[${#crates[@]}]="\"./tooling/bundler\""; fi
           if [ "${CLI}" == "true" ]; then crates[${#crates[@]}]="\"./tooling/cli\""; fi
-          echo ::set-output name=matrix::[$crates]
+          echo "matrix=matrix::[$crates]" >> "$GITHUB_ENV"
```
Comment (amrbashir) [inline .github/workflows/udeps.yml]: Shouldn't this be
```suggestion
          echo "matrix=[$crates]" >> "$GITHUB_ENV"
```

File: .github/workflows/udeps.yml [hunk 2]
```diff
@@ -162,4 +162,4 @@ jobs:
       - uses: actions-rs/cargo@v1
         with:
           command: udeps
-          args: --manifest-path ${{ matrix.path }}/Cargo.toml --all-targets --all-features
+          args: --manifest-path $matrix.path/Cargo.toml --all-targets --all-features
```
Comment (amrbashir) [inline .github/workflows/udeps.yml]: why remove the brackets?

**Review by amrbashir — CHANGES_REQUESTED** [review]

File: .github/workflows/udeps.yml (hunk 2 above)
Comment (martinfrances107) [reply .github/workflows/udeps.yml]: I have following a pattern from a tutorial I was following .. Ok I will push a change in a moment which follows the principle of make the smallest alteration.

File: .github/workflows/udeps.yml [hunk 3]
```diff
@@ -115,7 +115,7 @@ jobs:
           if [ "${UTILS}" == "true" ]; then crates[${#crates[@]}]="\"./core/tauri-utils\""; fi
           if [ "${BUNDLER}" == "true" ]; then crates[${#crates[@]}]="\"./tooling/bundler\""; fi
           if [ "${CLI}" == "true" ]; then crates[${#crates[@]}]="\"./tooling/cli\""; fi
-          echo ::set-output name=matrix::[$crates]
+          echo "matrix=[$crates]" >> "$GITHUB_ENV"
```
Comment (amrbashir) [inline .github/workflows/udeps.yml]: I should've noticed this earlier, we should be using `$GITHUB_OUTPUT`
```suggestion
          echo "matrix=[$crates]" >> "$GITHUB_OUTPUT"
```

**Review by amrbashir — APPROVED** [review]

**Pull request merged**